toolkit: enable downloads using managed identities (#14494)
microsoft/azurelinux: toolkit/docs/building/building.md

### Authentication

If supplying custom endpoints for source/SRPM/package servers, accessing these resources may require authentication.
Keys and certificates for TLS based authentication can be set using:

```bash
sudo make image CONFIG_FILE="./imageconfigs/core-efi.json" CA_CERT=/path/to/rootca.crt TLS_CERT=/path/to/user.crt TLS_KEY=/path/to/user.key
```

> For SRPM packing (i.e., for retrieving package sources), an Azure CLI based identity can be used to access authenticated Azure blob storages:

Reviewer: Clarity nit-pick:

```suggestion
For SRPM packing (i.e., for retrieving package sources), an Azure CLI login can be used to access Azure blob storages, which do not support anonymous access:
```

### Authentication

If supplying custom endpoints for source/SRPM/package servers, accessing these resources may require authentication.
Keys and certificates for TLS based authentication can be set using:

```bash
sudo make image CONFIG_FILE="./imageconfigs/core-efi.json" CA_CERT=/path/to/rootca.crt TLS_CERT=/path/to/user.crt TLS_KEY=/path/to/user.key
```

For SRPM packing (i.e., for retrieving package sources), Azure CLI login can be used to access authenticated Azure blob storages, which do not support anonymous access:
```bash
sudo make build-packages SOURCE_AUTH_MODE="azurecli"
```

> Using this ID requires prior `az login` with your managed identity ID.

Reviewer: Maybe?

```suggestion
Using this method requires prior `az login` with your managed identity ID.
```

| CA_CERT                       |                                                                                                          | CA cert to access the above resources, in addition to the system certificate store
| TLS_CERT                      |                                                                                                          | TLS cert to access the above resources
| TLS_KEY                       |                                                                                                          | TLS key to access the above resources

> | SOURCE_AUTH_MODE              |                                                                                                          | Enables managed identity based authentication for accessing package sources from protected Azure Blob Storages. This is useful when package sources are located in storage accounts prohibiting public access.

Reviewer: I'd also describe each of the available options.
